lib/inventory.ml

```diff
@@ -39,6 +39,17 @@ let loaded_inventory = ref false
 let inventory = Hashtbl.create 10
 let inventory_m = Mutex.create ()
 
+let write_to_inventory x =
+  (** at compile time, inventory_filename is assigned to a path whose parent folder does not exist on a host,
+      so not assigning this variable can give unhelpful error messages, e.g:
+      Unix.Unix_error(Unix.ENOENT, "open", "/usr/lib64/opamroot/ocaml-system/6be9d065-0ada-4ed8-a4af-cc388ea64913")
+    * we advise callers to check that they initialised inventory_filename *)
+  try
+    Unixext.write_string_to_file !inventory_filename x
+  with Unix.(Unix_error (ENOENT, _, _)) as e ->
+   Printf.eprintf "did you initialise 'inventory_filename'?";
```

Comment:
Do you have access to the debug logging methods here? I'm not sure whether this would go to the right place, or to /dev/null.